manim/mobject/three_dimensions.py

```diff
@@ -136,3 +137,70 @@ def generate_points(self):
         Cube.generate_points(self)
         for dim, value in enumerate(self.dimensions):
             self.rescale_to_fit(value, dim, stretch=True)
+
+
+class Honeycomb(Tiling):
+    """
+    Inherits from Tiling and works effectively the same, just adding
+    the third dimension.
+    To achieve this __init__ and apply_transforms are extended,
+    while tile_init_loop is overridden.
+    Since it's a 3D honeycomb also allows for tile_dictionary to be
+    alternatively called as cell_dictionary.
+    See Tiling for more details.
+
+    Parameters
+    ----------
+    tile_prototype : Mobject or function(x,y,z) that returns a Mobject
+    x_offset : nested list of Mobject methods and values
+    y_offset : nested list of Mobject methods and values
+    z_offset : nested list of Mobject methods and values
+    x_range : range
+    y_range : range
+    z_range : range
```

Comment:
apply my other comments from other reviews
note that, in all of them, i assumed that  x,y,z, are ints, but feel free to change that in the Callable signature

without changing things:
```suggestion
    tile_prototype : Union[:class:`~.Mobject`, Callable[[:class:`int`, :class:`int`, :class:`int`], :class:`~.Mobject`]]
    x_offset : Iterable[Iterable[Union[Callable[[:class:`int`, :class:`int`], :class:`~.Mobject`]]]]
    y_offset : Iterable[Iterable[Union[Callable[[:class:`int`, :class:`int`], :class:`~.Mobject`]]]]
    z_offset : Iterable[Iterable[Union[Callable[[:class:`int`, :class:`int`], :class:`~.Mobject`]]]]
    x_range : :class:`range`
    y_range : :class:`range`
    z_range : :class:`range`
```

If we change offsets to a list of lists of tuples, as proposed before:
```suggestion
    tile_prototype : Union[:class:`~.Mobject`, Callable[[:class:`int`, :class:`int`, :class:`int`], :class:`~.Mobject`]]
    x_offset : Iterable[Iterable[Tuple[Callable[[:class:`int`, :class:`int`], :class:`~.Mobject`]]]]
    y_offset : Iterable[Iterable[Tuple[Callable[[:class:`int`, :class:`int`], :class:`~.Mobject`]]]]
    z_offset : Iterable[Iterable[Tuple[Callable[[:class:`int`, :class:`int`], :class:`~.Mobject`]]]]
    x_range : :class:`range`
    y_range : :class:`range`
    z_range : :class:`range`
```